mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -114,10 +115,10 @@ final class QuantileDiscretizer @Since("1.6.0") (@Since("1.6.0") override val ui
     splits(0) = Double.NegativeInfinity
     splits(splits.length - 1) = Double.PositiveInfinity
 
-    val distinctSplits = splits.distinct
+    val distinctSplits = splits.filter(!_.isNaN).distinct
```

Comment:
Ah, I think this is a little bit at odds with the intent. We need to filter NaN before the data goes to approxQuantile. They should have no input on the quantiles. Then there's no need to filter out NaN from splits.

The message below can then remain unchanged from what it was before. This message is not related to the behavior of NaN.